### DISCUSSION TITLE
mimelist plugin: only gawk understands the current syntax

### DISCUSSION THREAD

**Discussion by mdhooge** [jarun/nnn#1815] [General]
Hello,

This took me time (and drove me crazy) to find why I could use the mimelist plugin on one computer and not on the other!

The mimelist plugin uses
```
find . | file -if- | grep "$mime" | awk -F: '{printf "%s\0", $1}' > "$NNN_PIPE"
``` 

but this is only working with Gnu Awk…

According to [this stackoverflow thread](https://stackoverflow.com/questions/9133315/how-can-i-output-null-terminated-strings-in-awk), I changed it to
```
find . | file -if- | grep "$mime" | awk -F: '{printf "%s%c", $1, 0}' >"$NNN_PIPE"
```

and now it works correctly.

I can provide a PR if needed.

Cheers

**Comment by KlzXS** [comment]
> I can provide a PR if needed.

Please do, but check all plugins for this.

**Comment by mdhooge** [comment]
PR submitted.

And BTW, using `file` is not the best option (besides it is found everywhere) as it is quite dumb at giving proper mime types for text files. For example, Python, Markdown, TOML, YAML are all detected as `text/plain`.

`mimetype` (installed as a dependency of `xdg-utils` in debian) is way better.

So the last line of the plugin could be:
```shell
if type mimetype >/dev/null 2>&1; then
    find . | mimetype -f - | grep "$mime" | awk -F: '{printf "%s%c", $1, 0}' > "$NNN_PIPE"
else
    find . | file -if- | grep "$mime" | awk -F: '{printf "%s%c", $1, 0}' > "$NNN_PIPE"
fi
```

https://github.com/mdhooge/nnn/commit/b46d608aa105db7cb026dc110dea5ce5a37a550e
